MNT Update `asv.conf.json` to get rid of last references to Python 2.7

DimitriPapadopoulos opened the pull request:
#### What does this implement/fix? Explain your changes.

* `six` is not used anymore
* Use newer version of Python in examples.

Also fix typo:
`platform` → `sys_platform`


#### Any other comments?

Not sure about this one, I have yet to understand the comment:
https://github.com/scikit-learn/scikit-learn/blob/85fb4dab9f76af8a06706da48dc7dd56916b105e/sklearn/utils/deprecation.py#L98-L100

DimitriPapadopoulos committed:
MNT Get rid of last references to Python 2.7

* six is not used anymore
* Use newer version of Python in examples.

Also fix typo:
platform → sys_platform

jeremiedbb commented:
Note that these references are in comments. They're here because this file was entirely copied from the asv config example file when it was configured for sklearn. That being said, I'm okay to remove these references since no one would use them in any case.

DimitriPapadopoulos commented:
My wrong then.

The upstream file https://github.com/airspeed-velocity/asv/blob/main/asv.conf.json has evolved on it own, and has removed references to Python 2. Perhaps I could sync with the upstream file.

DimitriPapadopoulos committed:
MNT asv: Restart from latest sample asv.conf.json

Fetch latest sample asv.conf.json from upstream:
https://github.com/airspeed-velocity/asv/blob/main/asv.conf.json

DimitriPapadopoulos committed:
MNT asv: Apply local changes to asv.conf.json

Apply local changes to sample asv.conf.json from upstream.

DimitriPapadopoulos committed:
MNT asv: platform → sys_platform

Fix typo from usptream:
https://github.com/airspeed-velocity/asv/pull/1477

DimitriPapadopoulos commented:
Sync'ed with latest [`asv.conf.json`](https://github.com/airspeed-velocity/asv/blob/main/asv.conf.json) and re-applied local changes.

jeremiedbb commented on asv_benchmarks/asv.conf.json:
```diff
@@ -4,30 +4,24 @@
 
     // The name of the project being benchmarked
-    "project": "scikit-learn",
+    "project": "asv",
```
```suggestion
    "project": "scikit-learn",
```

DimitriPapadopoulos replied:
I mixed up sync'ing and ended up overwriting my initial commit. Sorry about it.

jeremiedbb reviewed (commented):
Okay to sync with a more recent version of the config file

jeremiedbb reviewed (approved):
LGTM. Thanks
